### DISCUSSION TITLE
Adding GeigerLog Data?

### DISCUSSION THREAD

**Discussion by jhunolt** [poblabs/weewx-belchertown#780] [General]
Hi All, 
I currently have a WS-2902C and a GMC-500+ geiger counter.  I am running GeigerLog which give me a local web interface to the geiger counter data and added WeeWx (with Belchertown skin) so that I also have the weather station data locally.
What I would like to do is to incorporate either the text data (CPM, CPS, DR) or all the info in the screenshot.

I am trying to tell what the best approach is, should I look to just add text data from Geigerlog to WeeWX and display it with a WeeWx gauge, somehow link/add the GeigerLog site itself to the Belchertown skin, or move to MQTT which both apps support and somehow integrate the data there.

 Does anyone have an suggested methods or the pros and cons of different methods?

TIA,
Jon

![image](https://user-images.githubusercontent.com/25647650/165829827-e4fa9769-6709-45db-b0c6-b94b6c7e8098.png)

**Comment by jhunolt** [comment]
Hi All,
Just want to let anyone interested know that I did get this working so GeigerLog data is included in the WeeWX page.  It was done using the index_hook_after_station_info section and after ullix the developer of GeigerLog made some change to handle SSL and displaying the individual graphs.  Geiger counts are refreshed about every second.

Jon

![Screenshot 2022-06-04 10 53 20](https://user-images.githubusercontent.com/25647650/172020058-7e556d50-1423-4801-bc7b-869b2694fee2.png)

`


<td class="style24" style="width:720px"> 
<TH rowspan="2"><TH colspan="2">

<div style="position:relative;">
<br>	
<p>GQ GMC-500+ Data via GeigerLog</p>

<div align="left"><iframe src="https://xxxx:8080/widget_gauge?var=2" width="360" height="330 name="myFrame" frameborder="0"></iframe>

<iframe src="https://xxxx:8080/widget_gauge?var=3" width="360" height="330 name="myFrame" frameborder="0"></iframe></div>

<div align="left"><iframe src="https://xxxx:8080/widget_gauge?var=4" width="360" height="330 name="myFrame" frameborder="0"></iframe>

<iframe src="https://xxxx:8080/widget_gauge?var=5" width="360" height="330 name="myFrame" frameborder="0"></iframe></div>

<a href="https://xxxx:8080/mon" target="_blank" style="position:absolute; top:0; left:0; display:inline-block; width:500px; height:500px; z-index:5;"></a></div>


</div>
</TH>
`</td>`

**Comment by earlthesquirrel** [comment]
Just curious -- this operates off a independent device?
I ask because I extended the Weewx DB to include uSv/hr and simply plot via that.  I do like the gauge as well... that might nice on the front page of the site...

**Reply by jhunolt** [reply]
Hi, I am not sure what you mean by an independent device?  At the time I had one computer with everything connected to it and GeigerLog and WeeWX running there.

Is that what you meant?

Jon

**Reply by earlthesquirrel** [reply]
My weewx set up is running on a PC running a version of Linux.

All of my sensors are separate devices (i.e. WioNodes with sensors hooked to them, Arduinos with sensors, Adafruit Feathers with sensors, etc.).  My Geiger counter was deprived from the SwitchDocLabs Radsense project -- it communicates via 433Mhz to a Raspberry Pi that translates the data sent to it into MQTT messages which are then fed into weewx.

So, I was wondering if your Geiger counter is a sensor running independently from your weewx installation or not.

**Comment by jhunolt** [comment]
Hi,
My Geiger counter is a GMC-500Plus  https://www.gqelectronicsllc.com/comersus/store/comersus_viewItem.asp?idProduct=5631

It does WiFi or USB and also uploads automatically to the Geiger Counter World Map. 
https://www.gmcmap.com/

Then I was using GeigerLog https://github.com/ckuethe/geigerlog for viewing the data in WIndows.

So my goal was to incorporate the GeigerLog data into WeeWX and have just one web page to monitor.

Jon

**Reply by earlthesquirrel** [reply]
Interesting, so it's a stand-alone device with WiFi and USB.  I hadn't run into that one.  I've build 4 or 5 so far.
I'm playing with one of this right (at the office) : https://www.tindie.com/products/climateguard/dosimeter-with-i2c-radsens-2-arduino/

I build this one, but it has some component failure right now : https://www.adafruit.com/product/483 ( I got mine from mighty Ohm before Adafruit started making them)

And this one : https://www.dfrobot.com/product-2547.htm is what it currently feeding data (via this setup : https://www.switchdoc.com/2022/04/build-your-own-geiger-counter-radsense/ (though mine is inside and not running off solar/battery)

I just have a simple graph that I'm plotting data on.  I'm going to take a look at how you added your gauges and perhaps add that as well.

Each of my sensors was rather "independent" of my Weewx set up.  My weewx set up is running on an Intel PC board (running the Intel version of Raspbian - same linux as Raspberry Pi but an intel flavor -- saves me from mental hiccups moving from Pi's to it ;-) )    I've augmented the Geiger counters to send data via MQTT which is fed into my weewx...
